### PR TITLE
feat: clone object

### LINKED ISSUE #313: Object clone
Inside a single object view:

 * new `Clone` button
 * ask user for a new `title` using current as default with `-copy` suffix
 * a view form should appear like the one for new objects with all properties copied from original object and but empty `uname` (will be automatically created as default)

### PR DESCRIPTION
**Pull request by didoda**
Fixes #313

### CODE REVIEW

File: src/Controller/ModulesController.php [hunk 1]
```diff
@@ -353,6 +353,44 @@ public function saveJson() : void
         $this->set('_serialize', array_keys($response));
     }
 
+    /**
+     * Clone single resource.
```
Comment (stefanorosanelli) [inline src/Controller/ModulesController.php]: [superminor] here and below in params: `object` not `resource`

File: src/Template/Layout/js/app/app.js [hunk 2]
```diff
@@ -126,6 +126,24 @@ const _vueInstance = new Vue({
...
+        /**
+         * Clone object
+         * Prompt for title change
+         *
+         * @param {Event} e The event
+         * @return {void}
+         */
+        clone(e) {
+            const title = document.getElementById('title').value;
+            const cloneTitle = prompt(title, title + ' -copy');
```
Comment (stefanorosanelli) [inline src/Template/Layout/js/app/app.js]: User prompt should tell something more like:
`Please insert a new title on "{title}" clone"

File: src/Controller/ModulesController.php [hunk 3]
```diff
@@ -353,6 +353,44 @@ public function saveJson() : void
...
+     *
+     * @param string|int $id Resource ID.
+     * @return \Cake\Http\Response|null
+     */
+    public function clone($id) : ?Response
+    {
+        $this->viewBuilder()->setTemplate('view');
```
Comment (stefanorosanelli) [inline src/Controller/ModulesController.php]: unit test method is probably needed

File: src/Controller/ModulesController.php [hunk 4]
```diff
@@ -79,7 +79,7 @@ public function beforeRender(Event $event) : ?Response
     public function beforeFilter(Event $event) : ?Response
     {
         $actions = [
-            'delete', 'changeStatus', 'saveJson',
+            'delete', 'changeStatus', 'saveJson', 'clone'
```
Comment (stefanorosanelli) [inline src/Controller/ModulesController.php]: we should remove `'clone'` to avoid `403 Csrf token` error

File: src/Template/Layout/js/app/app.js [hunk 5]
```diff
@@ -126,6 +126,24 @@ const _vueInstance = new Vue({
...
+        /**
+         * Clone object
+         * Prompt for title change
+         *
+         * @param {Event} e The event
+         * @return {void}
+         */
+        clone(e) {
+            const title = document.getElementById('title').value;
+            const cloneTitle = prompt(`Please insert a new title on "${title}" clone`, title + ' -copy');
```
Comment (stefanorosanelli) [inline src/Template/Layout/js/app/app.js]: Is this message localized?

File: src/Locale/it_IT/default.po [hunk 6]
```diff
@@ -512,6 +548,18 @@ msgstr "mostra file"
 msgid "www.bedita.com"
 msgstr ""
 
+#: src/Template/Layout/js/app/app.js:138
+msgid "Please insert a new title on \"${ title }\" clone"
```
Comment (stefanorosanelli) [inline src/Locale/it_IT/default.po]: i'd expected standard placeholders like:
``` 
msgid "Please insert a new title on \"{0}\" clone"
```

File: src/Template/Layout/js/app/app.js (hunk 5 above)
Comment (didoda) [reply src/Template/Layout/js/app/app.js]: used ttag to translate...

File: src/Locale/it_IT/default.po (hunk 6 above)
Comment (didoda) [reply src/Locale/it_IT/default.po]: ttag syntax is different, as explained in ttag doc page https://github.com/ttag-org/ttag

**Pull request merged**